support/docker/build.sh

```diff
@@ -29,11 +29,136 @@ stderr.printf() {
 	printf "$@" >&2
 }
 
-_nix_docker() {
+
+_nix_container_manager() {
+
+	# Ask if the user wants to proceed as root
+	if (($(id -u) == 0)); 
+	then
+		stderr.printf "\nYou don't need to run this script as root.\n\nLear more on how run docker/podman in rootless mode:\n Docker: https://docs.docker.com/engine/security/rootless/\n Podman: https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md\n\n"
+		PS3="Running as root. Do you want to continue?: "
+
+		select yn in "Yes" "No"; do
+			case $yn in
+				Yes) 
+					stderr.printf "\nProceeding as root user\n"
+					break
+					;;
+				No)
+					stderr.printf "\nQuiting the build script...\n"
+					exit 0
+					;;
+			esac
+		done
+	fi
+
+	# TODO: Add selection menu of available boards located in boards dir
+
```

Comment:
```suggestion
```

Let's not add TODOs about changing the form of this script.

This script should provide the same semantics as `nix-build` does, since it's meant to allow non-Nix users to get acclimated into building with `nix-build`.

If we want a non-nix-build CLI helper to help people select what to build, it should be done externally to that script, and in a way that also works with `nix-build` without the container stuff. (note: I *do not know* if I want that, and that would be more for external potential contributors to discuss since obviously I'm well acquainted with `nix-build`.)

Reply:
Sounds good :+1: